### PR TITLE
Anchor.FM Gutenboarding passes along an episode id as well

### PR DESCRIPTION
**Pull request by mreishus**
#### Changes proposed in this Pull Request

* Anchor.FM Gutenboarding passes along an episode id as well

#### Testing instructions

* Visit  http://calypso.localhost:3000/new?&anchor_podcast=22b6608&anchor_episode=e324a06c-3148-43a4-85d8-34c0d8222138 in an incognito window
* Go through the gutenboarding process, creating a site and account
* Check the network tab when you finish, the episode id from the initial url will be passed to the wpcom site creation

* You can also add `console.log( { anchorFmEpisodeId, anchorFmPodcastId } );` to `use-step-navigation` if you'd like

#### Notes

* Is this another hex style ID or is it a guid with "-"?
  * Anchor asked us to allow any characters allowed in a URL.
* Abstract function `useAnchorParameter()` was used to cut down on the repetition between `useAnchorFmEpisodeId`, `useAnchorFmPodcastId`
* Each parameter has three names. What comes in the query string, what is stored in location state, and what is sent to wpcom.
  * Initial Query String: `?anchor_podcast=cafe1`
  * Location State: `anchorFmPodcastId`
  * Sent to WPCOM: `anchor_fm_podcast_id`.
  * If we can align some of these, the code will be simpler.  For example, changing the initial url to http://calypso.localhost:3000/new?anchorFmPodcastId=cafe1&anchorFmEpisodeId=f00d2 will allow `useAnchorParameter()` to be simplified.
* There is some other repetition in `use-step-navigation`.. Personally, I am fine with it as long as there are only 2 parameters, and would consider reworking if we have any more.

### CODE REVIEW

**Comment by mreishus** [comment]
Per p1607375826063900-slack-C017FUZGVAL, Episode GUIDs will need to have any character that is valid in a url to be allowed.

**Comment by retrofox** [comment]
Maybe I'm wrong but I think https://github.com/Automattic/jetpack/pull/17936 does something similar.
Just want to check we aren't overlapping work. :-)

**Comment by mreishus** [comment]
Two different parts of the process I believe, hopefully this makes sense

anchor-tickets

**Comment by retrofox** [comment]
> Two different parts of the process I believe, hopefully this makes sense
> 

`This might need to be added somewehere` ~= `Gutenframe`?

**Comment by allilevine** [comment]
> Abstract function `useAnchorParameter()` was used to cut down on the repetition between `useAnchorFmEpisodeId`, `useAnchorFmPodcastId`

Would it help prevent repetition to return the two ids together as an array or object?

> Each parameter has three names. What comes in the query string, what is stored in location state, and what is sent to wpcom.

And [in Jetpack](https://github.com/Automattic/jetpack/pull/17936) they're called `jetpack_anchor_podcast` and `jetpack_anchor_episode`. 😄

File: client/landing/gutenboarding/path.ts [hunk 1]
```diff
@@ -113,23 +115,70 @@ export function useIsAnchorFm(): boolean {
...
-	const sanitizePodcastId = ( id: string ) => id.replace( /[^a-zA-Z0-9]/g, '' );
+	const sanitize = ( id: string ) => id.replace( /[^a-zA-Z0-9]/g, '' );
+	return useAnchorParameter( {
+		queryParamName: 'anchor_podcast',
+		locationStateParamName: 'anchorFmPodcastId',
+		sanitize,
+	} );
+}
+
+// Returns the anchor podcast id. First looks in "location state",
```
Comment (allilevine) [inline client/landing/gutenboarding/path.ts]: Tiny edit, should this say episode id?

File: client/landing/gutenboarding/path.ts [hunk 2]
```diff
@@ -113,23 +115,70 @@ export function useIsAnchorFm(): boolean {
...
+	// Unreserved: A-Za-z0-9_.~-    (possibly % as a part of percent-encoding)
+	const sanitize = ( id: string ) => id.replace( /[^A-Za-z0-9_.\-~%]/g, '' );
+	return useAnchorParameter( {
+		queryParamName: 'anchor_episode',
+		locationStateParamName: 'anchorFmEpisodeId',
+		sanitize,
+	} );
+}
+
+/*
```
Comment (allilevine) [inline client/landing/gutenboarding/path.ts:140]: I wish there were more comments like this. 🏆

**Review by allilevine — APPROVED** [review]
Looks good! 🎉 I was able to pass along the episode id with these changes applied. I think we should align the id names and iterate on this to simplify it.

File: client/landing/gutenboarding/path.ts (hunk 1 above)
Comment (mreishus) [reply client/landing/gutenboarding/path.ts]: Fixed here [add/anchor-gutenboarding-episode-id 3e3c3066ec] Fix comment

**Pull request merged**